app/controllers/api/v1/experiences_controller.rb

```diff
@@ -62,20 +47,112 @@ def destroy
     private
       def set_user
         @user = User.find(params[:user_id])
+
+        if @user.id != current_user.id
+          raise ExceptionTypes::UnauthorizedError.new("You do not have access to modify the experiences of the user with ID #{@user.id}")
+        end
       end
 
-      # Use callbacks to share common setup or constraints between actions.
       def set_experience
         @experience = @user.experiences.find(params[:id]) if @user
       end
 
-      # Refactor these to be one method that takes a parameter for appropriate ID
-      def organization_experience_params
-        params.require(:experience).permit(:user_id, :organization_id, :start_date, :end_date, :title, :award, :current)
+      # Validate the request payload for creating a new experience
+      def validate_create_params
+        validate_ids
+        validate_dates
+        validate_current
+      end
+
+      # Validate the request payload for updating an existing experience
+      def validate_update_params
+        validate_dates
+        validate_current
+      end
+
+      # Validate that the ids in the payload belong to existing records
+      def validate_ids
+        organization_id = params[:organization_id] || params[:experience][:organization_id]
+        program_id = params[:program_id] || params[:experience][:program_id]
+        parent_org_id = params[:parent_org] || params[:experience][:parent_org]
+        
+        if !program_id && !organization_id
+          raise ExceptionTypes::BadRequestError.new("An experience must be associated with either a program or an organization")
+        end
+        if organization_id
+          Organization.find(organization_id)
```

Comment:
Why is this being called? It doesn't seem like we're doing anything with the result; is there some kind of side-effect that I don't know about?